pkgs/tools/backup/diskrsync/default.nix

```diff
@@ -1,27 +1,41 @@
-{ buildGoPackage, fetchFromGitHub, stdenv }:
+{ buildGoPackage, fetchFromGitHub, stdenv, openssh, runCommand, makeWrapper }:
 
-buildGoPackage rec {
+let
 
-  name = "${pname}-${version}";
-  pname = "diskrsync";
-  version = "unstable-2018-02-03";
+  diskrsync = buildGoPackage rec {
 
-  src = fetchFromGitHub {
-    owner = "dop251";
-    repo = pname;
-    rev = "2f36bd6e5084ce16c12a2ee216ebb2939a7d5730";
-    sha256 = "1rpfk7ds4lpff30aq4d8rw7g9j4bl2hd1bvcwd1pfxalp222zkxn";
-  };
+    name = "${pname}-${version}";
+    pname = "diskrsync";
+    version = "unstable-2018-02-03";
+
+    src = fetchFromGitHub {
+      owner = "dop251";
+      repo = pname;
+      rev = "2f36bd6e5084ce16c12a2ee216ebb2939a7d5730";
+      sha256 = "1rpfk7ds4lpff30aq4d8rw7g9j4bl2hd1bvcwd1pfxalp222zkxn";
+    };
+
+    goPackagePath = "github.com/dop251/diskrsync";
+    goDeps = ./deps.nix;
 
-  goPackagePath = "github.com/dop251/diskrsync";
-  goDeps = ./deps.nix;
+    meta = with stdenv.lib; {
+      description = "Rsync for block devices and disk images";
+      homepage = https://github.com/dop251/diskrsync;
+      license = licenses.mit;
+      platforms = platforms.all;
+      maintainers = with maintainers; [ jluttine ];
+    };
 
-  meta = with stdenv.lib; {
-    description = "Rsync for block devices and disk images";
-    homepage = https://github.com/dop251/diskrsync;
-    license = licenses.mit;
-    platforms = platforms.all;
-    maintainers = with maintainers; [ jluttine ];
   };
 
+in runCommand "diskrsync"
+{
+  buildInputs = [ makeWrapper ];
 }
+''
+  mkdir -p $out/bin
+  makeWrapper                  \
+    ${diskrsync}/bin/diskrsync \
+    $out/bin/diskrsync         \
+    --prefix PATH : ${openssh}/bin
```

Comment:
use `wrapProgram` instead in the `preFixup` phase of the original derivation instead in `runCommand`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/nrpl/default.nix#L32

Reply:
But that would cause unnecessary re-compilation of the Go package when just the runtime-only dependency SSH changes. Doesn't sound right to me...